python_modules/libraries/dagster-aws/dagster_aws/ecs/executor.py

```diff
@@ -0,0 +1,394 @@
+
+        # confusingly, run_task expects cpu and memory value as strings
+        self._cpu = str(cpu) if cpu else None
+        self._memory = str(memory) if memory else None
+
+        self._ephemeral_storage = ephemeral_storage
+        self._task_overrides = check.opt_mapping_param(task_overrides, "task_overrides")
+
+        current_task_metadata = get_current_ecs_task_metadata()
+        current_task = get_current_ecs_task(
+            self.ecs, current_task_metadata.task_arn, current_task_metadata.cluster
+        )
+
+        if run_launcher.use_current_ecs_task_config:
+            current_task_kwargs = get_task_kwargs_from_current_task(
+                self.ec2,
+                current_task_metadata.cluster,
+                current_task,
+            )
+        else:
+            current_task_kwargs = {}
+
+        run_launcher_kwargs = {**current_task_kwargs, **run_launcher.run_task_kwargs}
+
+        self._cluster_arn = current_task["clusterArn"]
+        self._task_definition_arn = current_task["taskDefinitionArn"]
+
+        # note: more kwargs will be pulled from the run launcher
+        # during launch_step
+        self._run_task_kwargs = {
+            **run_launcher_kwargs,
+            **run_task_kwargs,
+            "taskDefinition": current_task["taskDefinitionArn"],
+        }
+
+        # TODO: change launch_step to return task ARN
+        # this will be a breaking change so we need to wait fore a minor release
+        # to do this
+        self._launched_tasks = {}
+
+    def _get_run_task_kwargs(
+        self,
+        run: DagsterRun,
+        args: Sequence[str],
+        step_key: str,
+        step_tags: Mapping[str, str],
+        step_handler_context: StepHandlerContext,
+        container_context: EcsContainerContext,
+    ):
+        run_launcher = check.inst(
+            step_handler_context.instance.run_launcher,
+            EcsRunLauncher,
+            "ECS executor can only be enabled with the ECS run launcher",
+        )
+
+        run_task_kwargs = self._run_task_kwargs
+
+        kwargs_from_tags = step_tags.get("ecs/run_task_kwargs")
+        if kwargs_from_tags:
+            run_task_kwargs = {**run_task_kwargs, **json.loads(kwargs_from_tags)}
+
+        run_task_kwargs["tags"] = [
+            *run_task_kwargs.get("tags", []),
+            *[
+                {"key": k, "value": v}
+                for k, v in step_handler_context.dagster_run.dagster_execution_info.items()
+            ],
+            {"key": "dagster/step-key", "value": step_key},
+            {"key": "dagster/step-id", "value": self._get_step_id(step_handler_context)},
+        ]
+
+        if run.remote_job_origin:
+            run_task_kwargs["tags"] = [
+                *run_task_kwargs["tags"],
+                {
+                    "key": "dagster/code-location",
+                    "value": run.remote_job_origin.repository_origin.code_location_origin.location_name,
+                },
+            ]
```

Comment:
i think you want to call run_launcher.build_ecs_tags_for_run_task? so that it also incorporates tags that are set on the EcsContainerContext (for example, if in cloud they have set tags in their dagster_cloud.yaml, those should probably get propagated to the step tasks as well?)